### PR TITLE
Clarify documentation of custom scales

### PR DESCRIPTION
**Pull request by haberdashPI**
Based on discussion from #329

### CODE REVIEW

File: docs/gallery/gallery/scales/custom_scales.jl [hunk 1]
```diff
@@ -9,9 +9,21 @@ using AlgebraOfGraphics, CairoMakie
 using Colors
 set_aog_theme!() #src
 
-# Sometimes, there is no default palettes for a specific attribute. In that
-# case, the user can pass their own. TODO: allow legend to use custom attribute
-# of plot, such as the arrowhead or the arrowcolor and pass correct legend symbol.
+# You can customize the color palette used by a discrete scale using `pallettes`
+# in a call to `draw`. To customize a continuous scale, use `colormap` in a call
+# to [Visual](@ref). Using `pallettes` for a discrete scale goes from being
```
Comment (piever) [inline docs/gallery/gallery/scales/custom_scales.jl:14]:
```suggestion
# to [Visual](@ref). Using `palettes` for a discrete scale goes from being
```

File: docs/gallery/gallery/scales/custom_scales.jl [hunk 2]
```diff
@@ -9,9 +9,21 @@ using AlgebraOfGraphics, CairoMakie
...
+# optional to necessary when there is no default pallette for a particular scale
+# (e.g. arrowcolor). 
+
+# !!! note 
+#
+#    The reason discrete and continuous scales differ is because discrete scales
+#    require preprocessing by AlgebraOfGraphics. The `colormap` keyword comes
+#    directly from Makie, while `pallettes` is used by AlgebraOfGraphics
```
Comment (piever) [inline docs/gallery/gallery/scales/custom_scales.jl:22]:
```suggestion
#    directly from Makie, while `palettes` is used by AlgebraOfGraphics
```

**Review by piever — CHANGES_REQUESTED** [review]

**Comment by piever** [comment]
Thanks, this is very helpful! Looks good to me other than a couple of typos.

**Comment by haberdashPI** [comment]
I made a few changes, that I think will improve clarity, based on my slightly improved understanding. I'm going to also add a cross reference under the colormap section.

File: docs/gallery/gallery/scales/custom_scales.jl [hunk 3]
```diff
@@ -9,22 +9,26 @@ using AlgebraOfGraphics, CairoMakie
...
 
+# !!! note
+
+#    A related concept (from Makie) is a colormap, which maps a continuous space
+#    of numbers to a sequence of colors. For discrete colors you will want to
+#    employ a palette, not a colormap, because discrete values are mapped to
+#    colors within AlgebraOfGraphics. AlgebraOfGraphics doesn't directly handle
+#    continuous colors: this is a feature of the underlying Makie plots.
+#    Palettes and colormaps are also distinct in that a single colormap is
+#    defined for an entire figure, while palettes can vary by layer.
```
[the remark this answers is not in the document]
Comment (haberdashPI) [reply docs/gallery/gallery/scales/custom_scales.jl]: Ah, okay, I see. That makes sense. I'll revise.

File: docs/src/generated/visual.jl [hunk 4]
```diff
@@ -16,7 +16,13 @@ df = (x=randn(1000), y=randn(1000))
 plt = data(df) * mapping(:x, :y) * AlgebraOfGraphics.density(npoints=50)
 draw(plt * visual(Heatmap)) # plot as heatmap (the default)
 
+# We can specify a different colormap using the underlying Makie keyword
+# `colormap`. 
 #
+# !!! note
+#    For discrete scales you should not use a `colormap`;
+#    you will want to instead use the related notion of a [pallette](@ref Custom
+#    Scales)
```
Comment (piever) [inline docs/src/generated/visual.jl:25]:
```suggestion
#    you will want to instead use the related notion of a [palette](@ref Custom
#    Scales).
```

It also looks like Documenter cannot find the reference (doc build is failing).

File: docs/gallery/gallery/scales/custom_scales.jl [hunk 5]
```diff
@@ -9,9 +9,23 @@ using AlgebraOfGraphics, CairoMakie
...
+# 1 maps to green, 2 maps to red). Sometimes, there is no default pallette for a
+# specific attribute, and you will need to specify it manually, but there are
+# sensible default pallettes for many attributes. In either case, you can always
+# manually specify the pallette used for a particular attribute.
+
+# TODO: allow legend to use custom attribute of plot, such as the arrowhead or
+# the arrowcolor and pass correct legend symbol.
+
+# !!! note
+
```
Comment (piever) [inline docs/gallery/gallery/scales/custom_scales.jl:23]:
```suggestion
```

Just to be sure the conversion to markdown does not screw up the note formatting.

**Review by piever — CHANGES_REQUESTED** [review]

**Review by piever — APPROVED** [review]

**Pull request merged**